### DISCUSSION TITLE
"sudo RUST_LOG=info ./target/debug/bpfman-rpc --timeout=0" produces no logging output

### DISCUSSION THREAD

**Discussion by thediveo** [bpfman/bpfman#1079] [Q&A] [answered]
System is a Ubuntu 23.10, with bpfman-rpc build from sources.

Running `bpfman-rpc` as [per the instructions in the documentation](https://bpfman.io/main/getting-started/launching-bpfman/#run-as-a-long-lived-process):

```bash
sudo RUST_LOG=info ./target/debug/bpfman-rpc --timeout=0
```

does not produce _any_ logging output, and in particular nothing remotely similar to the example shown in the documentation. `bpfman-rpc` keeps running and has created its listening UDS, though.

What am I doing wrong so that I don't see any logging information?

Update: logging messages start to appear, albeit in a jerky fashion, when an example client connects to the service. Might there be some unwanted buffering going on...? As it is, this makes diagnosing very difficult.

**Comment by Billy99** [comment, marked as answer]
@thediveo You aren't doing anything wrong, we introduced a bug last week when running as a long lived process on the host. Logs aren't being initialized from `bpfman-rc` like they should be.

I am working on a fix now via #1077
